osu.Game/Audio/SampleInfo.cs

```diff
@@ -20,5 +22,16 @@ public SampleInfo(params string[] sampleNames)
         public IEnumerable<string> LookupNames => sampleNames;
 
         public int Volume { get; } = 100;
+
+        public override int GetHashCode()
+        {
+            return HashCode.Combine(sampleNames, Volume);
```

Comment:
this seems slightly dodge as `Equals` checks sequence equality while `GetHashCode()` will will do a reference-based hashcode. i think what might be better here is `StructuralComparisons.StructuralEqualityComparer.GetHashCode()` ([it exists!](https://docs.microsoft.com/en-us/dotnet/api/system.collections.structuralcomparisons.structuralequalitycomparer?view=net-5.0))